pkg/utils/errors/unwrap.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import "errors"
+
+// Unwrap and returns the root error.
+//
+// It will try to get Cause error (if err implements the 'causer'
+// interface) and will unwrap again; it will iterate through again
+// until gets the root error.
+func Unwrap(err error) error {
+	var (
+		cdone      bool
+		res, udone = unwrapErr(err)
+	)
+
+	for !(udone && cdone) {
+		res, cdone = unwrapCauser(res)
+		res, udone = unwrapErr(res)
+	}
```

Comment:
This is not going to work if we have `wrapped(causer(wrapped(causer(err))))` (or something similar). One option to implement it properly would be to iterate over wrapped and causer both at the same time, in a single function, instead of having 2 unwrap functions, one for wrapped and one for causer.